### ISSUE TITLE
lastModified does not throw error when path is a directory

### ISSUE THREAD

**Issue by ben182** [PlatformCommunity/flysystem-bunnycdn#28]
Hi,

I discovered an issue with the lastModified function if you pass a directory as the path. On other adapters like the ftp adatper, the function should throw an error like so: https://github.com/thephpleague/flysystem-ftp/blob/e59ae1b1370503a7925a75cc44f6f99cced1b2f0/FtpAdapter.php#L334. It will check if the file returned a lastModified date and since directories do not return that, it will fail. On the bunnycdn adapter it will not fail. It will only check for 404s or more than one file returned. This results also in a type error because lastModified expects FileAttributes to be returned but if you pass a directory, DirectoryAttributes will be returned.

Reproducible by just trying to call lastModified on a folder path.

What do you think about just checking for:
```php
if (is_a($object, DirectoryAttributes::class)) {
    throw UnableToReadFile::fromLocation($path, 'Path is a directory');
}
```
in the function lastModified?

**Comment by sifex** [comment]
Hi Ben,

This should now be resolved in v3.1.4 ❤️

https://github.com/PlatformCommunity/flysystem-bunnycdn/releases/tag/v3.1.4

**Comment by ben182** [comment]
Thanks so much
